src/core/lib/event_engine/posix_engine/posix_engine.h

```diff
@@ -138,14 +140,23 @@ class PosixEventEngine final : public PosixEventEngineWithFdSupport,
  public:
   class PosixDNSResolver : public EventEngine::DNSResolver {
    public:
-    ~PosixDNSResolver() override;
+#if GRPC_ARES == 1 && defined(GRPC_POSIX_SOCKET_TCP)
```

Comment:
On platforms that don't support ares, this class will have an auto-generated public default constructor. Making a private, no-op default constructor, either in non-ares builds or just always having it, would prevent this from being instantiated on platforms that don't support ares. It's a bit of defensive programming, but I think it conveys the intention a bit better at compile time.

Reply:
Good point! I explicitly deleted the default constructor which should convey the same meaning here.